cpukit/score/src/schedulerstrongapa.c

```diff
@@ -6,248 +6,566 @@
+  
+  ret=NULL; //To remove compiler warning. 
+  //ret would always point to the node with the lowest priority
+  //node unless the affinity of filter_base is NULL.
+       
+  filter_node = _Scheduler_strong_APA_Node_downcast( filter_base );
+  
+  max_priority = 300;//Max (Lowest) priority encountered so far.
+  
+  cpu_max = _SMP_Get_processor_maximum();
+  _Chain_Initialize_empty(&Queue);
+  
+  for ( cpu_index = 0 ; cpu_index < cpu_max ; ++cpu_index ) { 
+    if( (  _Processor_mask_Is_set( &filter_node->affinity , cpu_index)
+     && visited[ cpu_index ] == false ) ) { 
+          //Checks if the thread_CPU is in the affinity set of the node
+      Per_CPU_Control *cpu = _Per_CPU_Get_by_index( cpu_index );
+      if( _Per_CPU_Is_processor_online( cpu ) ) {
+        Qcpu = rtems_malloc( sizeof(CPU) ); //No errornum returned in case of failure
+        Qcpu->cpu=*cpu;
+  
+        _Chain_Initialize_node( &Qcpu->node );
+        _Chain_Append_unprotected( &Queue, &Qcpu->node );
+         //Insert cpu in the Queue
+        visited[ cpu_index ]=true;
+      }
+    }
+  }
+  
+  while( !_Chain_Is_empty( &Queue) ) {
+    Qcpu = (CPU*) _Chain_Get_first_unprotected( &Queue );
+    curr_CPU = &Qcpu->cpu;
+    curr_thread = curr_CPU->executing;
+
+    curr_node = (Scheduler_Node *) _Chain_First( &curr_thread->Scheduler.Scheduler_nodes );
+    
+    //How to check if the thread is not participating
+    //in helping on this processor?
+  
+    curr_priority = _Scheduler_Node_get_priority( curr_node );
+    curr_priority = SCHEDULER_PRIORITY_PURIFY( curr_priority );   
+      
+    if(curr_priority < max_priority) {
+      ret = curr_node;
+      max_priority = curr_priority;
+    }
+    
+    Scurr_node = _Scheduler_strong_APA_Node_downcast( curr_node ); 
+    if( !curr_thread->is_idle ) {
+      for ( cpu_index = 0 ; cpu_index < cpu_max ; ++cpu_index ) {
+        if( _Processor_mask_Is_set( &Scurr_node->affinity , cpu_index ) ) { 
+          //Checks if the thread_CPU is in the affinity set of the node
+          Per_CPU_Control *cpu = _Per_CPU_Get_by_index( cpu_index );
+          if( _Per_CPU_Is_processor_online( cpu ) && visited[ cpu_index ] == false ) {
+          
+            Qcpu = rtems_malloc( sizeof(CPU) );
+	    Qcpu->cpu=*cpu;
+	      
+	    _Chain_Initialize_node( &Qcpu->node );
+	    _Chain_Append_unprotected( &Queue, &Qcpu->node ); 
+	    //Insert the cpu in the affinty set of curr_thread in the Queue
+            visited[ cpu_index ]=true;
+          }
+        }  
+      }
+    }
+  }
+  
+  Priority_Control  filter_priority;
+  filter_priority = _Scheduler_Node_get_priority( filter_base );
+  filter_priority = SCHEDULER_PRIORITY_PURIFY( filter_priority );   
+ 
+  if( ret->Priority.value < filter_priority ) {
+    //Lowest priority task found has higher priority 
+    // than filter_base.
+    //So, filter_base remains unassigned
+    //No task shifting. 
+  }  
+ 
+  else {
+    //Backtrack on the path from
```

Comment:
similar comments as above, change this to the if { } branch, and elaborate a plan for how to achieve the backtracking.